lib/mongoose_push_web/router.ex

```diff
@@ -5,9 +5,26 @@ defmodule MongoosePushWeb.Router do
     plug(:accepts, ["json"])
   end
 
+  pipeline :swagger_json do
+    plug(:accepts, ["json"])
+    plug(OpenApiSpex.Plug.PutApiSpec, module: MongoosePushWeb.ApiSpec)
+  end
+
   scope "/api", MongoosePushWeb do
     pipe_through(:api)
 
     post("/dummy", DummyController, :handle)
   end
+
+  scope "/" do
+    pipe_through(:swagger_json)
+
+    get("/swagger.json", OpenApiSpex.Plug.RenderSpec, [])
+  end
+
+  scope "/v1", MongoosePushWeb do
+    pipe_through(:api)
+
+    post("/notification", APIv1Controller, :handle)
```

Comment:
How about:

```suggestion
    post("/notification", APIv1.PushController, :send)
```

Reply:
Controllers always should group operations one the same objects (notification), while handlers should represent operations on this object (send?).